app/plugin_api.go

```diff
@@ -57,6 +63,39 @@ func (api *PluginAPI) LoadPluginConfiguration(dest interface{}) error {
 	}
 }
 
+func (api *PluginAPI) ValidateConfig(config map[string]interface{}) bool {
```

Comment:
Can you please clarify why plugin config validation needs to be a public method?

Reply:
will make private, no reason I can see why it should be public

Reply:
since `func (m *Manifest) ValidateConfig(config map[string]interface{}) error` is now called from `Manifest` within `func (env *Environment) Activate(id string, config *model.Config) (manifest *model.Manifest, activated bool, reterr error)` I don't think i can make it private. I am close to checking in the PR once unit tests are done. I will try to adjust the visibility of `func (m *Manifest) ValidateConfig(config map[string]interface{}) error` during unit testing